### PR TITLE
Remove Lecture.session

### PR DESCRIPTION
**Pull request by ronnix**

### CODE REVIEW

File: repondeur/zam_repondeur/models/texte.py [hunk 1]
```diff
@@ -85,6 +85,12 @@ class Texte(Base):
         "date_depot",
     )
 
+    @property
+    def session_str(self) -> Optional[str]:
+        if self.session is None:
+            return None
```
Comment (davidbgk) [inline repondeur/zam_repondeur/models/texte.py]: If we don't return an empty string we will have `None` in URLs too?

Comment (ronnix) [reply repondeur/zam_repondeur/models/texte.py]: For AN, `Texte.session` is `None` but we have `Texte.legislature`. For Sénat, it's the opposite. So we should never have None in the Sénat fetch URLs.

Comment (davidbgk) [reply repondeur/zam_repondeur/models/texte.py]: 👌

File: repondeur/zam_repondeur/fetch/senat/derouleur.py [hunk 2]
```diff
@@ -127,22 +127,23 @@ def _parse_derouleur_data(data_iter: Iterable[Any]) -> List[DiscussionDetails]:
 
 
 def derouleur_urls(lecture: Lecture) -> Iterator[str]:
+    assert lecture.texte.session_str is not None  # mypy hint
```
Comment (davidbgk) [inline repondeur/zam_repondeur/fetch/senat/derouleur.py]: I'm surprised this line doesn't generate an error on the bandit side.

Comment (ronnix) [reply repondeur/zam_repondeur/fetch/senat/derouleur.py]: Yup

Comment (ronnix) [reply repondeur/zam_repondeur/fetch/senat/derouleur.py]: Rebase manually, solved the conflicts, and bandit was indeed unhappy with the asserts.

**Pull request merged**